presto-hive/src/main/java/com/facebook/presto/hive/metastore/MetastoreUtil.java

```diff
@@ -257,4 +268,43 @@ public static void verifyCanDropColumn(ExtendedHiveMetastore metastore, String d
             throw new PrestoException(NOT_SUPPORTED, "Cannot drop the only non-partition column in a table");
         }
     }
+
+    public static FileSystem getFileSystem(HdfsEnvironment hdfsEnvironment, HdfsContext context, Path path)
+    {
+        try {
+            return hdfsEnvironment.getFileSystem(context, path);
+        }
+        catch (IOException e) {
+            throw new PrestoException(HIVE_FILESYSTEM_ERROR, format("Error getting file system. Path: %s", path), e);
+        }
+    }
+
+    public static void renameFile(FileSystem fileSystem, Path source, Path target)
+    {
+        try {
+            if (fileSystem.exists(target) || !fileSystem.rename(source, target)) {
+                throw new PrestoException(HIVE_FILESYSTEM_ERROR, getRenameErrorMessage(source, target));
+            }
+        }
+        catch (IOException e) {
+            throw new PrestoException(HIVE_FILESYSTEM_ERROR, getRenameErrorMessage(source, target), e);
+        }
+    }
+
+    private static String getRenameErrorMessage(Path source, Path target)
+    {
+        return format("Error moving data files from %s to final location %s", source, target);
+    }
+
+    public static void waitForListenableFutures(List<ListenableFuture<?>> listenableFutures)
+    {
+        ListenableFuture<?> listenableFutureAggregate = whenAllSucceed(listenableFutures).call(() -> null, directExecutor());
+        try {
+            getFutureValue(listenableFutureAggregate);
```

Comment:
I think you want to do `getFutureValue(listenableFutureAggregate, PrestoException.class)`